module/metrics/transaction.go

```diff
@@ -136,6 +152,18 @@ func (tc *TransactionCollector) ScriptExecuted(dur time.Duration, size int) {
 	}).Observe(float64(dur.Milliseconds()))
 }
 
+func (tc *TransactionCollector) ScriptExecutionErrorOnArchiveNode(blockID flow.Identifier, scriptHash string) {
+	// record the execution error along with blockID and scriptHash for Archive node
+	tc.scriptExecutionErrorArchiveNode.With(prometheus.Labels{"block_id": blockID.String(), "script": scriptHash}).Inc()
```

Comment:
labels should be mostly static, with a small set of unique values. otherwise, it causes performance issues on the prometheus server.

Rather than using block and script hash, how about using a label for executor with either `execution`/`archive`. You could also have a code for the error type if there are existing buckets.

Reply:
`execution || archive`  is better, I'll use that